packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -140,6 +140,13 @@ impl Abigen {
                         pub fn _get_wallet(&self) -> LocalWallet {
                             self.wallet.clone()
                         }
+
+                        pub fn _with_wallet(&self, mut wallet: LocalWallet) -> Result<Self, SDKError> {
+                           let provider = self.wallet.get_provider()?;
+                           wallet.set_provider(provider.clone());
+
+                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet })
+                        }
```

Comment:
We should document somewhere that the `Provider` inside `wallet` will be ignored and superseded by the `Provider` originally used to create the contract instance. 

Users could try and create a wallet that holds a different `Provider`, pass it to `_with_wallet()` and get some weird errors because that `Provider` has been ignored.

I don't think this should be a use case at all (i.e., testing stuff with a single instance on two different wallets on two different providers), but we should at least cover this in the docs somewhere.

Reply:
Docs have been updated. Let me know if it is clear enough.